src/app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-server.cpp

```diff
@@ -360,9 +360,9 @@ CHIP_ERROR SetValveLevel(EndpointId ep, DataModel::Nullable<Percent> level, Data
     if (!isDelegateNull(delegate))
     {
         DataModel::Nullable<Percent> cLevel = delegate->HandleOpenValve(level);
-        if (HasFeature(ep, ValveConfigurationAndControl::Feature::kLevel))
+        if (!cLevel.IsNull())
```

Comment:
It is not guaranteed that the delegate will do the right thing and returning null if it doesn't support the Level feature. So keeping the feature map check seems alright to me.

Reply:
Added back in the feature check - should be checking both.